src/limbo/bayes_opt/boptimizer.hpp

```diff
@@ -113,6 +114,9 @@ namespace limbo {
                     } else {
                         _model.add_sample(this->_samples.back(), this->_observations.back(), Params::bayes_opt_boptimizer::noise());
                     }
+                    if (this->_current_iteration % Params::bayes_opt_boptimizer::hp_period() == 0
+                        || Params::bayes_opt_boptimizer::hp_period() == -1)
```

Comment:
Agreed. This is probably what I had in mind.

-- JBM

On Sat, May 14, 2016 at 12:02 AM Konstantinos Chatzilygeroudis <
notifications@github.com> wrote:

> In src/limbo/bayes_opt/boptimizer.hpp
> <https://github.com/resibots/limbo/pull/100#discussion_r63257155>:
>
> > @@ -113,6 +114,9 @@ namespace limbo {
> >                      } else {
> >                          _model.add_sample(this->_samples.back(), this->_observations.back(), Params::bayes_opt_boptimizer::noise());
> >                      }
> > +                    if (this->_current_iteration % Params::bayes_opt_boptimizer::hp_period() == 0
> > +                        || Params::bayes_opt_boptimizer::hp_period() == -1)
>
> @jbmouret <https://github.com/jbmouret> What's the use of the second
> check (hp_period()==-1), since *x%1* is *always equal to 0*? So, if
> someone wants to optimize in every iteration, he should set the hp_period
> to 1.
>
> We should use hp_period() == -1 to not optimize at all. What do you think?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/resibots/limbo/pull/100/files/1f81a2e80c85bfe43a2fe10bf19416b719bfdaa3#r63257155>
>